File: handlerclaw/lib/features/api-keys/data/mappers/api_key_mapper.dart
import 'package:handlerclaw/features/api-keys/data/dto/api_key_dto.dart';
import 'package:handlerclaw/features/api-keys/domain/entities/api_key_entity.dart';

class ApiKeyMapper {
  static ApiKeyEntity toEntity(ApiKeyDto dto) {
    return ApiKeyEntity(
      id: dto.id,
      name: dto.name,
      keyPreview: dto.keyPreview,
      status: _mapStatus(dto.status),
      createdAt: dto.createdAt,
      updatedAt: dto.updatedAt,
      plainKey: dto.plainKey,
    );
  }

  static ApiKeyStatus _mapStatus(String status) {
    switch (status.toUpperCase()) {
      case 'ACTIVE':
        return ApiKeyStatus.ACTIVE;
      case 'REVOKED':
        return ApiKeyStatus.REVOKED;
      case 'EXPIRED':
        return ApiKeyStatus.EXPIRED;
      default:
        return ApiKeyStatus.ACTIVE;
    }
  }
}

File: api/src/modules/(jwt-auth)/api-keys/infrastructure/mappers/api-key.mapper.ts
import type { ApiKey } from "../../../../../lib/generated/prisma/client.js";
import type { IApiKey } from "../../domain/entities/api-key.entity.js";

export function toApiKeyEntity(data: ApiKey): IApiKey {
  return {
    id: data.id,
    user_id: data.userId,
    name: data.name,
    key_preview: data.keyPreview,
    key_full: data.keyFull,
    status: data.status,
    created_at: data.createdAt,
    updated_at: data.updatedAt,
    deleted_at: data.deletedAt,
  };
}
